timelock-impl/src/test/java/com/palantir/atlasdb/timelock/LockTokenConvertingTimelockServiceTest.java

```diff
@@ -77,4 +78,38 @@ public void identifiersPreservedOnRepeatedConversions() {
             assertThat(tokenV2.getRequestId()).isEqualTo(TEST_UUID);
         }
     }
+
+    @Test
+    public void convertsBigIntegersCorrectly() {
+        int iterations = 100_000;
+        int bitLength = 128;
+        Random random = new Random(0);
+        for (int i = 0; i < iterations; i++) {
+            BigInteger randomBigInteger = new BigInteger(bitLength, random);
+            LockRefreshToken lockRefreshToken = new LockRefreshToken(randomBigInteger, Long.MIN_VALUE);
+            assertConversionFromAndToLegacyPreservesId(lockRefreshToken);
+        }
+    }
+
+    @Test
+    public void convertsNegativeBigIntegersCorrectly() {
+        int iterations = 100_000;
+        int bitLength = 128;
+        Random random = new Random(1);
+        for (int i = 0; i < iterations; i++) {
+            BigInteger randomBigInteger = new BigInteger(bitLength, random).multiply(new BigInteger("-1"));
+            LockRefreshToken lockRefreshToken = new LockRefreshToken(randomBigInteger, Long.MIN_VALUE);
+            assertConversionFromAndToLegacyPreservesId(lockRefreshToken);
+        }
+    }
+
+    private void assertConversionFromAndToLegacyPreservesId(LockRefreshToken lockRefreshToken) {
+        LockTokenV2 initialToken = new LegacyTimelockService.LockRefreshTokenV2Adapter(lockRefreshToken);
+        LockTokenV2 serializable = LockTokenConvertingTimelockService.makeSerializable(initialToken);
+        LockTokenV2 reconverted = LockTokenConvertingTimelockService.castToAdapter(serializable);
```

Comment:
i think we actually want to test that the bigint contained within this adapter is the same. i.e. `reconverted.equals(initialToken)`